File: SocialNetwork.Domain/Entity/FriendRequest.cs

using SocialNetwork.Domain.Abstractions;

namespace SocialNetwork.Domain.Entity
{
    public class FriendRequest : BaseAuditableEntity
    {
        public string Status { get; set; }
        public string SenderId { get; set; }
        public User Sender { get; set; }
        public string ReceiverId { get; set; }
        public User Receiver { get; set; }
    }
}

File: SocialNetwork.WebApp/src/types/friendRequest.ts
import { FriendRequestStatus } from "../enums/friend-request";
import { UserResource } from "./user";

export type FriendRequestResource = {
    id: string;
    sender: UserResource;
    senderId: string;
    receiverId: string;
    dateCreated: Date;
    dateUpdated: Date;
    status: FriendRequestStatus;
}

